src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntry.cs

```diff
@@ -12,27 +12,43 @@ namespace Microsoft.Health.Dicom.Core.Features.CustomTag
     /// </summary>
     public class CustomTagEntry : IEquatable<CustomTagEntry>
     {
+        public CustomTagEntry(string path, string vR, CustomTagLevel level, CustomTagStatus status)
+        {
+            Path = path;
+            VR = vR;
+            Level = level;
+            Status = status;
+        }
+
+        public CustomTagEntry(CustomTagStoreEntry storedEntry)
+        {
+            Path = storedEntry.Path;
+            VR = storedEntry.VR;
+            Level = storedEntry.Level;
+            Status = storedEntry.Status;
+        }
+
         /// <summary>
         /// Path of this tag. Normally it's composed of groupid and elementid.
         /// E.g: 00100020 is path of patient id.
         /// </summary>
-        public string Path { get; set; }
+        public string Path { get; }
```

Comment:
I got failure when calling add custom tag API without "set;". The reason is that CustomTagEntry is used by Asp.net to deserialize and store input.